src/Libraries/Microsoft.Extensions.AI.Abstractions/UsageDetails.cs

```diff
@@ -19,8 +22,25 @@ public class UsageDetails
     /// <summary>Gets or sets the total number of tokens used to produce the response.</summary>
     public int? TotalTokenCount { get; set; }
 
-    /// <summary>Gets or sets additional properties for the usage details.</summary>
-    public AdditionalPropertiesDictionary? AdditionalProperties { get; set; }
+    /// <summary>Gets additional usage counts.</summary>
+    [JsonObjectCreationHandling(JsonObjectCreationHandling.Populate)]
+    public IDictionary<string, long> AdditionalCounts { get; } = new Dictionary<string, long>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This differs from how we've done AdditionalProperties everywhere else. Why have this be get-only with populate rather than being a nullable property with a setter?

Reply:
> Is it because this is using the interface and so this is the only way to guarantee what comparer is used

Yes that's exactly it. Or in fact you could generalize the concern to say that allowing settable `IDictionary`-type properties anywhere is dubious if these are intended as exchange types (or we could say "collaboration types") that multiple unrelated pieces of software are meant to use as a shared workspace. Whoever writes the dictionary first gets to pick its semantics, which might have arbitrary behaviors unlike a standard `Dictionary<T, U>`.

Swapping this to use `AdditionalPropertiesDictionary` (which becomes generic) makes sense. I'll see if it works out.

Reply:
This is done. It works out OK.

The only drawback is we need to either:

 * Explicitly surface `AdditionalPropertiesDictionary<object?>` in a bunch of places on the public API, which makes it look and feel more complicated
 * Or, also derive a subclass `AdditionalPropertiesDictionary` with TValue=object, which is totally fine but then the base type has to be unsealed

In practice I don't know of a problem from having the generic base class be unsealed. It doesn't even have any `virtual` members. So I think keeping the public API surface looking clean in all other places outweighs this, and hence I have added a nongeneric subclass too.